scripts/testing/programs/match_template/test_template_reproducibility.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python3
+"""
+Template Matching Reproducibility Test
+
+This script runs the cisTEM template matching GPU binary three times using the same input data
+and parameters, then analyzes the reproducibility by comparing the resulting MIP images.
+
+It uses the Apoferritin dataset for testing, and saves the MIP (Maximum Intensity Projection)
+files to a temporary directory with unique filenames for each replicate.
+
+The script loads the MIP images using the mrcfile package and compares them using numpy to
+measure pixel similarity between replicates with various metrics.
+"""
+
+import annoying_hack
+
+from os.path import join, exists
+from os import makedirs
+import cistem_test_utils.args as tmArgs
+import cistem_test_utils.make_tmp_runfile as mktmp
+import cistem_test_utils.run_job as runner
+import cistem_test_utils.temp_dir_manager as temp_dir_manager
+import mrcfile
+import numpy as np
+import tempfile
+import os
+import shutil
+import sys
+import re
+
+# By default the "_gpu" suffix will be added unless the --old-cistem flag is used
+# or the --cpu flag is used
+wanted_binary_name = 'match_template'
+
+
+def extract_threshold_value(hist_file_path):
```

Comment:
This is something that might be useful in other tests so lets move it to its own module in cistem_test_utils and then import it in this file for use.